react-app/src/components/ButtonsModal/index.js

```diff
@@ -0,0 +1,53 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import BaseModal from 'components/BaseModal';
+
+import styles from './index.module.scss';
+
+export default class ButtonsModal extends PureComponent {
+  static propTypes = {
+    /**
+     * Whenever the modal is visible or not
+     */
+    visible: PropTypes.bool.isRequired,
+
+    /**
+     * The message that the modal will show
+     */
+    message: PropTypes.string.isRequired,
+
+    /**
+     * The primary button (a Button) that the element will show
+     */
+    primaryButton: PropTypes.element.isRequired,
+
+    /**
+     * The (optional) secondary button (a Button) that the element will show
+     */
+    secondaryButton: PropTypes.element,
+  };
+
+  static defaultProps = { secondaryButton: null }
+
+  render() {
+    const content = (
+      <div className={styles.container}>
+        <div>
+          {this.props.message}
+        </div>
+        <div className={styles.buttonsContainer}>
+          {this.props.secondaryButton}
+          {this.props.primaryButton}
+        </div>
+      </div>
+    );
+
+    return (
+      <BaseModal
+        visible={this.props.visible}
+        content={content}
```

Comment:
If you were to use the `children` prop on `BaseModal` instead of `content`, you will be able to write the component composition like this:

```jsx
      <BaseModal visible={this.props.visible}>
        {content}
      </BaseModal>
```

Which is more flexible and represent better the hierarchy of components.
See here: https://reactjs.org/docs/composition-vs-inheritance.html#containment

Reply:
In fact, using it like that will allow you to use the content directly inside the `BaseModal` in a better looking way than if you were to do so using a `content` prop. Here's how it looks:

```jsx
      <BaseModal visible={this.props.visible}>
        <div className={styles.container}>
          <div>
            {this.props.message}
          </div>
          <div className={styles.buttonsContainer}>
            {this.props.secondaryButton}
            {this.props.primaryButton}
          </div>
        </div>
      </BaseModal>
```

Reply:
Great tip! Done